x-pack/plugins/ml/server/models/bucket_span_estimator/bucket_span_estimator.js

```diff
@@ -321,54 +326,89 @@ export function estimateBucketSpanFactory(callWithRequest) {
+          includeDefaults: true,
+          filterPath: '*.*max_buckets'
         })
-        .catch((resp) => {
-          reject(resp);
-        });
+          .then((settings) => {
+            if (typeof settings !== 'object' || typeof settings.defaults !== 'object') {
+              reject('Unable to retrieve cluster setting search.max_buckets');
+            }
+
+            const maxBuckets = parseInt(settings.defaults['search.max_buckets']);
+
+            const runEstimator = (splitFieldValues = []) => {
+              const bucketSpanEstimator = new BucketSpanEstimator(
+                formConfig,
+                splitFieldValues,
+                maxBuckets
+              );
+
+              bucketSpanEstimator.run()
+                .then((resp) => {
+                  resolve(resp);
+                })
+                .catch((resp) => {
+                  reject(resp);
+                });
+            };
+
+            // a partition has been selected, so we need to load some field values to use in the
+            // bucket span tests.
+            if (formConfig.splitField !== undefined) {
+              getRandomFieldValues(formConfig.index, formConfig.splitField, formConfig.query)
+                .then((splitFieldValues) => {
+                  runEstimator(splitFieldValues);
+                })
+                .catch((resp) => {
+                  reject(resp);
+                });
+            } else {
+              // no partition field selected or we're in the single metric config
+              runEstimator();
+            }
+          })
+          .catch((resp) => {
+            reject(resp);
+          });
+      }
+
+      if (isSecurityDisabled(server)) {
+        getBucketSpanEstimation();
+      } else {
+        // if security is enabled, check that the user has permission to
+        // view jobs before calling getBucketSpanEstimation.
+        // getBucketSpanEstimation calls the 'cluster.getSettings' endpoint as the internal user
+        // and so could give the user access to more information than
+        // they are entitled to.
+        const body = {
+          cluster: [
+            'cluster:monitor/xpack/ml/job/get',
+            'cluster:monitor/xpack/ml/job/stats/get',
+            'cluster:monitor/xpack/ml/datafeeds/get',
+            'cluster:monitor/xpack/ml/datafeeds/stats/get'
+          ]
+        };
+        callWithRequest('ml.privilegeCheck', { body })
+          .then((resp) => {
+            if (resp.cluster['cluster:monitor/xpack/ml/job/get'] &&
+              resp.cluster['cluster:monitor/xpack/ml/job/stats/get'] &&
+              resp.cluster['cluster:monitor/xpack/ml/datafeeds/get'] &&
+              resp.cluster['cluster:monitor/xpack/ml/datafeeds/stats/get']) {
+              getBucketSpanEstimation();
+            } else {
+              // if the user doesn't have permission to create jobs
+              // return a 403
+              reject(Boom.forbidden());
```

Comment:
this Boom response will not be interpreted correctly by the calling function and the route.
Maybe just a simple `reject('Permission denied')` would be enough.
Not that a user with no job permissions will ever get into a situation where they are able to call this endpoint. Unless they are calling manually with curl